src/views/home/home.hooks.ts

```diff
@@ -0,0 +1,311 @@
+import { ProfilePreviewData } from '@/components/ProfilePreviewCard';
+import { IndexerSDK, Listing } from '@/modules/indexer';
+import { PublicKey } from '@solana/web3.js';
+import { HomeData } from 'pages';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useHomeQuery } from 'src/graphql/indexerTypes';
+import {
+  BuyNowListingFragment,
+  HomeQuery,
+  MarketplacePreviewFragment,
+  ProfilePreviewFragment,
+} from 'src/graphql/indexerTypes.ssr';
+import { FeedItem } from '../alpha/feed.utils';
+import { FeaturedBuyNowListingsData, ListingPreviewData } from './FeaturedBuyNowListingsSection';
+import { FeaturedMarketplacesData, MarketplacePreviewData } from './FeaturedMarketplacesSection';
+import { FeaturedProfilesData } from './FeaturedProfilesSection';
+import { HeroSectionData } from './HeroSection';
+
+export interface QueryContext<T> {
+  data?: T | undefined;
+  loading: boolean;
+  error?: Error | undefined;
+  refetch: () => void;
+}
+
+//TODO remove once other profiles have enough followers to preclude this one in the backend
+const DISALLOWED_PROFILES: string[] = ['ho1aVYd4TDWCi1pMqFvboPPc3J13e4LgWkWzGJpPJty'];
+
+export function useHomeQueryWithTransforms(
+  userWallet: PublicKey | null,
+  featuredProfileLimit: number,
+  featuredBowNowLimit: number,
+  feedEventsLimit: number,
+  featuredAuctionsLimit: number
+): QueryContext<HomeData> {
+  const queryContext = useHomeQuery({
+    variables: {
+      userWallet: userWallet,
+      featuredProfileLimit: featuredProfileLimit,
+      featuredBuyNowLimit: featuredBowNowLimit,
+      feedEventsLimit: feedEventsLimit,
+    },
+  });
+
+  const featuredAuctionContext: QueryContext<Listing[]> = usePrepareFeaturedAuctions(featuredAuctionsLimit);
+
+  const loading: boolean = queryContext.loading || featuredAuctionContext.loading;
+
+  const data: HomeData | undefined = useMemo(() => {
+    let result: HomeData | undefined;
+    if (!loading && queryContext.called && !queryContext.error) {
+      result = {
+        feedEvents: transformHeroSectionData(queryContext.data?.feedEvents),
+        featuredProfiles: transformFeaturedProfiles(queryContext.data?.followWallets),
+        featuredMarketplaces: transformFeaturedMarketplaces(
+          queryContext.data?.featuredMarketplace0,
```

Comment:
I'd prefer this as a list, but no biggie

Reply:
oh, whoops, I forgot I added a query for multiple marketplaces :D